defaultNatsService.go

```diff
@@ -0,0 +1,409 @@
+package natsio
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/nats-io/nats.go"
+	"github.com/orchestd/dependencybundler/interfaces/configuration"
+	"github.com/orchestd/dependencybundler/interfaces/credentials"
+	"github.com/orchestd/dependencybundler/interfaces/log"
+	. "github.com/orchestd/servicereply"
+	"github.com/orchestd/servicereply/status"
+	"go.uber.org/fx"
+	"reflect"
+	"time"
+)
+
+var connect = func(natsUrl string, options ...nats.Option) (NatsConnection, error) {
+	return nats.Connect(natsUrl, options...)
+}
+
+func NewNatsServiceWithoutConnection(logger log.Logger) NatsService {
+	return &defaultNatsService{
+		logger:        logger,
+		subscriptions: map[string]*nats.Subscription{},
+	}
+}
+
+func NewNatsServiceWithBasicAuth(lc fx.Lifecycle, config configuration.Config, logger log.Logger, credentials credentials.CredentialsGetter) NatsService {
+	return getDefaultService(lc, config, logger, func(serviceName string) nats.Option {
+		natsUser := credentials.GetCredentials().NatsUser
+		if natsUser == "" {
+			panic("could not get credentials by key NatsUser")
```

Comment:
funny you changed to could not get 
but all other errors (including one you just added...) are "cant"....